# Logistic Regression result

## origin dataset

best:   

!python logistic_reg.py --dataset origin --fraud_weight 10  
Precision rate:  0.7238095238095238  
Recall rate: 0.8085106382978723  
Accuracy：0.9991748885221726  

In [78]:
!python logistic_reg.py --dataset origin --fraud_weight 0.5

/Users/wenqi/anaconda3/lib/python3.7/site-packages/sklearn/linear_model/logistic.py:433: FutureWarning: Default solver will be changed to 'lbfgs' in 0.22. Specify a solver to silence this warning.
  FutureWarning)
TP count:    46
FP count:    10
TN count:    56858
FN count:    48
Precision rate:  0.8214285714285714
Recall rate: 0.48936170212765956


In [79]:
!python logistic_reg.py --dataset origin --fraud_weight 1

/Users/wenqi/anaconda3/lib/python3.7/site-packages/sklearn/linear_model/logistic.py:433: FutureWarning: Default solver will be changed to 'lbfgs' in 0.22. Specify a solver to silence this warning.
  FutureWarning)
TP count:    56
FP count:    13
TN count:    56855
FN count:    38
Precision rate:  0.8115942028985508
Recall rate: 0.5957446808510638


In [80]:
!python logistic_reg.py --dataset origin --fraud_weight 5

/Users/wenqi/anaconda3/lib/python3.7/site-packages/sklearn/linear_model/logistic.py:433: FutureWarning: Default solver will be changed to 'lbfgs' in 0.22. Specify a solver to silence this warning.
  FutureWarning)
TP count:    73
FP count:    26
TN count:    56842
FN count:    21
Precision rate:  0.7373737373737373
Recall rate: 0.776595744680851


In [117]:
# best one
!python logistic_reg.py --dataset origin --fraud_weight 10

/Users/wenqi/anaconda3/lib/python3.7/site-packages/sklearn/linear_model/logistic.py:433: FutureWarning: Default solver will be changed to 'lbfgs' in 0.22. Specify a solver to silence this warning.
  FutureWarning)
time consumed: 0.005075009000000019
TP count:    76
FP count:    29
TN count:    56839
FN count:    18
Precision rate:  0.7238095238095238
Recall rate: 0.8085106382978723
Accuracy：0.9991748885221726


In [82]:
!python logistic_reg.py --dataset origin --fraud_weight 100

/Users/wenqi/anaconda3/lib/python3.7/site-packages/sklearn/linear_model/logistic.py:433: FutureWarning: Default solver will be changed to 'lbfgs' in 0.22. Specify a solver to silence this warning.
  FutureWarning)
TP count:    81
FP count:    161
TN count:    56707
FN count:    13
Precision rate:  0.3347107438016529
Recall rate: 0.8617021276595744


In [83]:
!python logistic_reg.py --dataset origin --fraud_weight 1000

/Users/wenqi/anaconda3/lib/python3.7/site-packages/sklearn/linear_model/logistic.py:433: FutureWarning: Default solver will be changed to 'lbfgs' in 0.22. Specify a solver to silence this warning.
  FutureWarning)
TP count:    89
FP count:    2403
TN count:    54465
FN count:    5
Precision rate:  0.03571428571428571
Recall rate: 0.9468085106382979


In [84]:
!python logistic_reg.py --dataset origin --fraud_weight 10000

/Users/wenqi/anaconda3/lib/python3.7/site-packages/sklearn/linear_model/logistic.py:433: FutureWarning: Default solver will be changed to 'lbfgs' in 0.22. Specify a solver to silence this warning.
  FutureWarning)
TP count:    91
FP count:    17605
TN count:    39263
FN count:    3
Precision rate:  0.005142405063291139
Recall rate: 0.9680851063829787


## subsample dataset
best:

!python logistic_reg.py --dataset subsample --fraud_weight 0.0005  
Precision rate:  0.5793650793650794  
Recall rate: 0.776595744680851  
Accuracy：0.9987008883115059  

In [95]:
!python logistic_reg.py --dataset subsample --fraud_weight 0.0005

/Users/wenqi/anaconda3/lib/python3.7/site-packages/sklearn/linear_model/logistic.py:433: FutureWarning: Default solver will be changed to 'lbfgs' in 0.22. Specify a solver to silence this warning.
  FutureWarning)
TP count:    73
FP count:    53
TN count:    56815
FN count:    21
Precision rate:  0.5793650793650794
Recall rate: 0.776595744680851
Accuracy：0.9987008883115059


In [94]:
!python logistic_reg.py --dataset subsample --fraud_weight 0.001

/Users/wenqi/anaconda3/lib/python3.7/site-packages/sklearn/linear_model/logistic.py:433: FutureWarning: Default solver will be changed to 'lbfgs' in 0.22. Specify a solver to silence this warning.
  FutureWarning)
TP count:    74
FP count:    76
TN count:    56792
FN count:    20
Precision rate:  0.49333333333333335
Recall rate: 0.7872340425531915
Accuracy：0.9983146659176293


In [85]:
!python logistic_reg.py --dataset subsample --fraud_weight 0.005

/Users/wenqi/anaconda3/lib/python3.7/site-packages/sklearn/linear_model/logistic.py:433: FutureWarning: Default solver will be changed to 'lbfgs' in 0.22. Specify a solver to silence this warning.
  FutureWarning)
TP count:    75
FP count:    134
TN count:    56734
FN count:    19
Precision rate:  0.3588516746411483
Recall rate: 0.7978723404255319


In [86]:
!python logistic_reg.py --dataset subsample --fraud_weight 0.01

/Users/wenqi/anaconda3/lib/python3.7/site-packages/sklearn/linear_model/logistic.py:433: FutureWarning: Default solver will be changed to 'lbfgs' in 0.22. Specify a solver to silence this warning.
  FutureWarning)
TP count:    77
FP count:    154
TN count:    56714
FN count:    17
Precision rate:  0.3333333333333333
Recall rate: 0.8191489361702128


In [87]:
!python logistic_reg.py --dataset subsample --fraud_weight 0.1

/Users/wenqi/anaconda3/lib/python3.7/site-packages/sklearn/linear_model/logistic.py:433: FutureWarning: Default solver will be changed to 'lbfgs' in 0.22. Specify a solver to silence this warning.
  FutureWarning)
TP count:    77
FP count:    228
TN count:    56640
FN count:    17
Precision rate:  0.25245901639344265
Recall rate: 0.8191489361702128


In [88]:
!python logistic_reg.py --dataset subsample --fraud_weight 0.5

/Users/wenqi/anaconda3/lib/python3.7/site-packages/sklearn/linear_model/logistic.py:433: FutureWarning: Default solver will be changed to 'lbfgs' in 0.22. Specify a solver to silence this warning.
  FutureWarning)
TP count:    83
FP count:    679
TN count:    56189
FN count:    11
Precision rate:  0.1089238845144357
Recall rate: 0.8829787234042553


In [89]:
!python logistic_reg.py --dataset subsample --fraud_weight 1

/Users/wenqi/anaconda3/lib/python3.7/site-packages/sklearn/linear_model/logistic.py:433: FutureWarning: Default solver will be changed to 'lbfgs' in 0.22. Specify a solver to silence this warning.
  FutureWarning)
TP count:    87
FP count:    1351
TN count:    55517
FN count:    7
Precision rate:  0.06050069541029207
Recall rate: 0.925531914893617


In [90]:
!python logistic_reg.py --dataset subsample --fraud_weight 5

/Users/wenqi/anaconda3/lib/python3.7/site-packages/sklearn/linear_model/logistic.py:433: FutureWarning: Default solver will be changed to 'lbfgs' in 0.22. Specify a solver to silence this warning.
  FutureWarning)
TP count:    90
FP count:    7378
TN count:    49490
FN count:    4
Precision rate:  0.012051419389394752
Recall rate: 0.9574468085106383


In [91]:
!python logistic_reg.py --dataset subsample --fraud_weight 10

/Users/wenqi/anaconda3/lib/python3.7/site-packages/sklearn/linear_model/logistic.py:433: FutureWarning: Default solver will be changed to 'lbfgs' in 0.22. Specify a solver to silence this warning.
  FutureWarning)
TP count:    92
FP count:    13749
TN count:    43119
FN count:    2
Precision rate:  0.006646918575247453
Recall rate: 0.9787234042553191


# SVM result

## origin dataset

best: 
!python SVM.py --dataset origin --fraud_weight 5  
Precision rate:  0.8021978021978022
Recall rate: 0.776595744680851
Accuracy：0.9993153330290369

In [96]:
!python SVM.py --dataset origin --fraud_weight 0.001

/Users/wenqi/anaconda3/lib/python3.7/site-packages/sklearn/svm/base.py:931: ConvergenceWarning: Liblinear failed to converge, increase the number of iterations.
  "the number of iterations.", ConvergenceWarning)
TP count:    0
FP count:    0
TN count:    56868
FN count:    94
Traceback (most recent call last):
  File "SVM.py", line 40, in <module>
    prec_recall(y_val, val_label_predict)
  File "/Users/wenqi/home/Scalable-Credit-Card-Fraud-Detection-in-Stream-Systems/sklearn/prec_recall.py", line 34, in prec_recall
    print('Precision rate:  {}'.format(TP/(TP+FP)))
ZeroDivisionError: division by zero


In [97]:
!python SVM.py --dataset origin --fraud_weight 0.005

/Users/wenqi/anaconda3/lib/python3.7/site-packages/sklearn/svm/base.py:931: ConvergenceWarning: Liblinear failed to converge, increase the number of iterations.
  "the number of iterations.", ConvergenceWarning)
TP count:    0
FP count:    0
TN count:    56868
FN count:    94
Traceback (most recent call last):
  File "SVM.py", line 40, in <module>
    prec_recall(y_val, val_label_predict)
  File "/Users/wenqi/home/Scalable-Credit-Card-Fraud-Detection-in-Stream-Systems/sklearn/prec_recall.py", line 34, in prec_recall
    print('Precision rate:  {}'.format(TP/(TP+FP)))
ZeroDivisionError: division by zero


In [98]:
!python SVM.py --dataset origin --fraud_weight 0.1

/Users/wenqi/anaconda3/lib/python3.7/site-packages/sklearn/svm/base.py:931: ConvergenceWarning: Liblinear failed to converge, increase the number of iterations.
  "the number of iterations.", ConvergenceWarning)
TP count:    43
FP count:    14
TN count:    56854
FN count:    51
Precision rate:  0.7543859649122807
Recall rate: 0.4574468085106383
Accuracy：0.9988588883817282


In [99]:
!python SVM.py --dataset origin --fraud_weight 0.3

/Users/wenqi/anaconda3/lib/python3.7/site-packages/sklearn/svm/base.py:931: ConvergenceWarning: Liblinear failed to converge, increase the number of iterations.
  "the number of iterations.", ConvergenceWarning)
TP count:    63
FP count:    12
TN count:    56856
FN count:    31
Precision rate:  0.84
Recall rate: 0.6702127659574468
Accuracy：0.9992451107756047


In [100]:
!python SVM.py --dataset origin --fraud_weight 0.5

/Users/wenqi/anaconda3/lib/python3.7/site-packages/sklearn/svm/base.py:931: ConvergenceWarning: Liblinear failed to converge, increase the number of iterations.
  "the number of iterations.", ConvergenceWarning)
TP count:    67
FP count:    12
TN count:    56856
FN count:    27
Precision rate:  0.8481012658227848
Recall rate: 0.7127659574468085
Accuracy：0.9993153330290369


In [109]:
!python SVM.py --dataset origin --fraud_weight 1

/Users/wenqi/anaconda3/lib/python3.7/site-packages/sklearn/svm/base.py:931: ConvergenceWarning: Liblinear failed to converge, increase the number of iterations.
  "the number of iterations.", ConvergenceWarning)
TP count:    69
FP count:    15
TN count:    56853
FN count:    25
Precision rate:  0.8214285714285714
Recall rate: 0.7340425531914894
Accuracy：0.9992977774656788


In [115]:
!python SVM.py --dataset origin --fraud_weight 5

/Users/wenqi/anaconda3/lib/python3.7/site-packages/sklearn/svm/base.py:931: ConvergenceWarning: Liblinear failed to converge, increase the number of iterations.
  "the number of iterations.", ConvergenceWarning)
TP count:    73
FP count:    18
TN count:    56850
FN count:    21
Precision rate:  0.8021978021978022
Recall rate: 0.776595744680851
Accuracy：0.9993153330290369


In [101]:
!python SVM.py --dataset origin --fraud_weight 10

/Users/wenqi/anaconda3/lib/python3.7/site-packages/sklearn/svm/base.py:931: ConvergenceWarning: Liblinear failed to converge, increase the number of iterations.
  "the number of iterations.", ConvergenceWarning)
TP count:    41
FP count:    119
TN count:    56749
FN count:    53
Precision rate:  0.25625
Recall rate: 0.43617021276595747
Accuracy：0.9969804431024192


## subsample dataset
best:
recision rate:  0.7047619047619048
Recall rate: 0.7872340425531915

In [116]:
!python SVM.py --dataset subsample --fraud_weight 0.0001

/Users/wenqi/anaconda3/lib/python3.7/site-packages/sklearn/svm/base.py:931: ConvergenceWarning: Liblinear failed to converge, increase the number of iterations.
  "the number of iterations.", ConvergenceWarning)
TP count:    74
FP count:    22
TN count:    56846
FN count:    20
Precision rate:  0.7708333333333334
Recall rate: 0.7872340425531915
Accuracy：0.9992626663389628


In [103]:
#### this one Recall rate: 0.7872340425531915
!python SVM.py --dataset subsample --fraud_weight 0.0005

/Users/wenqi/anaconda3/lib/python3.7/site-packages/sklearn/svm/base.py:931: ConvergenceWarning: Liblinear failed to converge, increase the number of iterations.
  "the number of iterations.", ConvergenceWarning)
TP count:    74
FP count:    69
TN count:    56799
FN count:    20
Precision rate:  0.5174825174825175
Recall rate: 0.7872340425531915
Accuracy：0.9984375548611355


In [104]:
!python SVM.py --dataset subsample --fraud_weight 0.001

/Users/wenqi/anaconda3/lib/python3.7/site-packages/sklearn/svm/base.py:931: ConvergenceWarning: Liblinear failed to converge, increase the number of iterations.
  "the number of iterations.", ConvergenceWarning)
TP count:    74
FP count:    86
TN count:    56782
FN count:    20
Precision rate:  0.4625
Recall rate: 0.7872340425531915
Accuracy：0.998139110284049


In [63]:
!python SVM.py --dataset subsample --fraud_weight 0.01

/Users/wenqi/anaconda3/lib/python3.7/site-packages/sklearn/svm/base.py:931: ConvergenceWarning: Liblinear failed to converge, increase the number of iterations.
  "the number of iterations.", ConvergenceWarning)
TP count:    74
FP count:    89
TN count:    56779
FN count:    20
Precision rate:  0.4539877300613497
Recall rate: 0.7872340425531915


In [50]:
!python SVM.py --dataset subsample --fraud_weight 0.1

/Users/wenqi/anaconda3/lib/python3.7/site-packages/sklearn/svm/base.py:931: ConvergenceWarning: Liblinear failed to converge, increase the number of iterations.
  "the number of iterations.", ConvergenceWarning)
TP count:    77
FP count:    304
TN count:    56564
FN count:    17
Precision rate:  0.2020997375328084
Recall rate: 0.8191489361702128


In [51]:
!python SVM.py --dataset subsample --fraud_weight 0.5

/Users/wenqi/anaconda3/lib/python3.7/site-packages/sklearn/svm/base.py:931: ConvergenceWarning: Liblinear failed to converge, increase the number of iterations.
  "the number of iterations.", ConvergenceWarning)
TP count:    79
FP count:    751
TN count:    56117
FN count:    15
Precision rate:  0.09518072289156626
Recall rate: 0.8404255319148937


In [45]:
!python SVM.py --dataset subsample --fraud_weight 1

/Users/wenqi/anaconda3/lib/python3.7/site-packages/sklearn/svm/base.py:931: ConvergenceWarning: Liblinear failed to converge, increase the number of iterations.
  "the number of iterations.", ConvergenceWarning)
TP count:    88
FP count:    2140
TN count:    54728
FN count:    6
Precision rate:  0.03949730700179533
Recall rate: 0.9361702127659575


In [46]:
!python SVM.py --dataset subsample --fraud_weight 5

/Users/wenqi/anaconda3/lib/python3.7/site-packages/sklearn/svm/base.py:931: ConvergenceWarning: Liblinear failed to converge, increase the number of iterations.
  "the number of iterations.", ConvergenceWarning)
TP count:    89
FP count:    6506
TN count:    50362
FN count:    5
Precision rate:  0.013495072024260804
Recall rate: 0.9468085106382979


In [118]:
!python SVM.py --dataset subsample --fraud_weight 10

/Users/wenqi/anaconda3/lib/python3.7/site-packages/sklearn/svm/base.py:931: ConvergenceWarning: Liblinear failed to converge, increase the number of iterations.
  "the number of iterations.", ConvergenceWarning)
time consumed: 0.0041208219999999685
TP count:    88
FP count:    8861
TN count:    48007
FN count:    6
Precision rate:  0.009833500949826796
Recall rate: 0.9361702127659575
Accuracy：0.8443348197043643


In [6]:
!python RandomForest.py

time:	0.008481062999999955
result:	[1 0 1 0 1 0 1 0 1 0]
TP count:    89
FP count:    14
TN count:    80
FN count:    5
Precision rate:  0.8640776699029126
Recall rate: 0.9468085106382979
Mean score:  0.898936170212766


# kNN result

In [112]:
!python knn.py --dataset origin

time:	103.176996203
result:	[0 0 0 0 0 0 0 0 0 0]
TP count:    69
FP count:    13
TN count:    56855
FN count:    25
Precision rate:  0.8414634146341463
Recall rate: 0.7340425531914894
Accuracy：0.9993328885923949


In [113]:
!python knn.py --dataset subsample

time:	2.053465976
result:	[0 0 0 0 0 0 0 0 0 0]
TP count:    81
FP count:    910
TN count:    55958
FN count:    13
Precision rate:  0.08173562058526741
Recall rate: 0.8617021276595744
Accuracy：0.98379621502054


# Random Forest result

## origin

In [121]:
!python RandomForest.py --dataset origin --fraud_weight 1 --n_estimators 100 --max_depth 10

time consumed: 0.27396105100000057
TP count:    70
FP count:    10
TN count:    56858
FN count:    24
Precision rate:  0.875
Recall rate: 0.7446808510638298
Accuracy：0.999403110845827


In [131]:
!python RandomForest.py --dataset origin --fraud_weight 1 --n_estimators 10 --max_depth 10

time consumed: 0.03745109699999993
TP count:    70
FP count:    7
TN count:    56861
FN count:    24
Precision rate:  0.9090909090909091
Recall rate: 0.7446808510638298
Accuracy：0.9994557775359011


In [135]:
!python RandomForest.py --dataset origin --fraud_weight 10 --n_estimators 100 --max_depth 10

time consumed: 0.3013742210000032
TP count:    71
FP count:    9
TN count:    56859
FN count:    23
Precision rate:  0.8875
Recall rate: 0.7553191489361702
Accuracy：0.9994382219725431


In [134]:
!python RandomForest.py --dataset origin --fraud_weight 10 --n_estimators 10 --max_depth 10

time consumed: 0.034107109999999
TP count:    72
FP count:    8
TN count:    56860
FN count:    22
Precision rate:  0.9
Recall rate: 0.7659574468085106
Accuracy：0.9994733330992591


In [136]:
!python RandomForest.py --dataset origin --fraud_weight 100 --n_estimators 10 --max_depth 10

time consumed: 0.03597039000000102
TP count:    73
FP count:    14
TN count:    56854
FN count:    21
Precision rate:  0.8390804597701149
Recall rate: 0.776595744680851
Accuracy：0.999385555282469


In [141]:
# best
!python RandomForest.py --dataset origin --fraud_weight 1000 --n_estimators 10 --max_depth 10

time consumed: 0.03224829000000007
TP count:    73
FP count:    17
TN count:    56851
FN count:    21
Precision rate:  0.8111111111111111
Recall rate: 0.776595744680851
Accuracy：0.9993328885923949


In [144]:
!python RandomForest.py --dataset origin --fraud_weight 5000 --n_estimators 10 --max_depth 10

time consumed: 0.03831567699999994
TP count:    78
FP count:    110
TN count:    56758
FN count:    16
Precision rate:  0.4148936170212766
Recall rate: 0.8297872340425532
Accuracy：0.9977879990168884


In [138]:
!python RandomForest.py --dataset origin --fraud_weight 10000 --n_estimators 10 --max_depth 10

time consumed: 0.03407728799999976
TP count:    78
FP count:    177
TN count:    56691
FN count:    16
Precision rate:  0.3058823529411765
Recall rate: 0.8297872340425532
Accuracy：0.9966117762719006


## subsample

In [130]:
!python RandomForest.py --dataset subsample --fraud_weight 1 --n_estimators 10 --max_depth 10

time consumed: 0.035028114999999915
TP count:    82
FP count:    1364
TN count:    55504
FN count:    12
Precision rate:  0.056708160442600276
Recall rate: 0.8723404255319149
Accuracy：0.9758435448193532


In [123]:
!python RandomForest.py --dataset subsample --fraud_weight 10 --n_estimators 100 --max_depth 10

time consumed: 0.3034473059999999
TP count:    85
FP count:    522
TN count:    56346
FN count:    9
Precision rate:  0.1400329489291598
Recall rate: 0.9042553191489362
Accuracy：0.9906779958568871


In [125]:
!python RandomForest.py --dataset subsample --fraud_weight 100 --n_estimators 100 --max_depth 10

time consumed: 0.3189671690000001
TP count:    89
FP count:    5093
TN count:    51775
FN count:    5
Precision rate:  0.017174835970667695
Recall rate: 0.9468085106382979
Accuracy：0.9105017380007725


In [127]:
!python RandomForest.py --dataset subsample --fraud_weight 1000 --n_estimators 100 --max_depth 10

time consumed: 0.298156187
TP count:    92
FP count:    25598
TN count:    31270
FN count:    2
Precision rate:  0.003581159984429739
Recall rate: 0.9787234042553191
Accuracy：0.5505775780344792


In [126]:
!python RandomForest.py --dataset subsample --fraud_weight 10000 --n_estimators 100 --max_depth 10

time consumed: 0.306445608
TP count:    93
FP count:    27922
TN count:    28946
FN count:    1
Precision rate:  0.0033196501873996075
Recall rate: 0.9893617021276596
Accuracy：0.5097960043537797
